apps/gdal_footprint_lib.cpp

```diff
@@ -116,12 +117,177 @@ struct GDALFootprintOptions
+
+    argParser->add_argument("-split_polys")
+        .flag()
+        .store_into(psOptions->bSplitPolys)
+        .help(_("Split multipolygons into several features each one with a "
+                "single polygon."));
+
+    argParser->add_argument("-convex_hull")
+        .flag()
+        .store_into(psOptions->bConvexHull)
+        .help(_("Compute the convex hull of the (multi)polygons."));
+
+    argParser->add_argument("-densify")
+        .metavar("<value>")
+        .scan<'g', double>()
+        .store_into(psOptions->dfDensifyDistance)
+        .help(_("The specified value of this option is the maximum distance "
+                "between 2 consecutive points of the output geometry. "));
+
+    argParser->add_argument("-simplify")
+        .metavar("<value>")
+        .scan<'g', double>()
+        .store_into(psOptions->dfSimplifyTolerance)
+        .help(_("The specified value of this option is the tolerance used to "
+                "merge consecutive points of the output geometry."));
+
+    argParser->add_argument("-min_ring_area")
+        .metavar("<value>")
+        .scan<'g', double>()
+        .store_into(psOptions->dfMinRingArea)
+        .help(_("The specified value of this option is the minimum area of a "
+                "ring to be considered."));
+
+    // Note: no store_into (requires post validation)
+    argParser->add_argument("-max_points")
+        .metavar("<value>|unlimited")
+        .default_value("100")
+        .help(_("The maximum number of points in the output geometry."));
+
+    if (psOptionsForBinary)
+    {
+        argParser->add_quiet_argument(&psOptionsForBinary->bQuiet);
+        argParser->add_open_options_argument(
+            psOptionsForBinary->aosOpenOptions);
+    }
+
+    argParser->add_output_format_argument(psOptions->osFormat);
+
+    argParser->add_argument("-location_field_name")
+        .metavar("<field_name>")
+        .default_value("location")
+        .store_into(psOptions->osLocationFieldName)
+        .help(_("Specifies the name of the field in the resulting vector "
+                "dataset where the path of the input dataset will be stored."));
+
+    argParser->add_argument("-no_location")
+        .flag()
+        .store_into(psOptions->bClearLocation)
+        .help(_("Turns off the writing of the path of the input dataset as a "
+                "field in the output vector dataset."));
+
+    argParser->add_argument("-write_absolute_path")
+        .flag()
+        .store_into(psOptions->bAbsolutePath)
+        .help(_("Enables writing the absolute path of the input dataset."));
+
+    argParser->add_layer_creation_options_argument(psOptions->aosLCO);
+
+    argParser->add_argument("-dsco")
+        .metavar("<NAME>=<VALUE>")
+        .append()
+        .action([psOptions](const std::string &s)
+                { psOptions->aosDSCO.AddString(s.c_str()); })
+        .help(_("Dataset creation option (format specific)."));
```

Comment:
that one could be a candidate for a helper in GDALArgumentParser . It is at least used by gdal_contour and ogr2ogr

Reply:
I had the same thought, I will do it in a follow-up PR.